addons/account/wizard/account_secure_entries_wizard.py

```diff
@@ -0,0 +1,268 @@
+from datetime import timedelta
+
+from odoo import Command, api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.osv import expression
+from odoo.tools import format_list
+from odoo.tools.misc import format_date
+
+
+class AccountSecureEntries(models.TransientModel):
+    """
+    This wizard is used to secure journal entries (with a hash)
+    """
+    _name = 'account.secure.entries.wizard'
+    _description = 'Secure Journal Entries'
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.company,
+    )
+    country_code = fields.Char(
+        related="company_id.account_fiscal_country_id.code",
+    )
+    hash_date = fields.Date(
+        string='Hash All Entries',
+        required=True,
+        help="The selected Date",
+    )
+    chains_to_hash_with_gaps = fields.Json(
+        compute='_compute_data',
+    )
+    max_hash_date = fields.Date(
+        string='Max Hash Date',
+        compute='_compute_max_hash_date',
+        help="Highest Date such that all posted journal entries prior to (including) the date are secured. Only journal entries after the hard lock date are considered.",
+    )
+    unreconciled_bank_statement_line_ids = fields.Many2many(
+        compute='_compute_data',
+        comodel_name='account.bank.statement.line',
+        help="All unreconciled bank statement lines before the selected date.",
+    )
+    not_hashable_unlocked_move_ids = fields.Many2many(
+        compute='_compute_data',
+        comodel_name='account.move',
+        help="All unhashable moves before the selected date that are not protected by the Hard Lock Date",
+    )
+    move_to_hash_ids = fields.Many2many(
+        compute='_compute_data',
+        comodel_name='account.move',
+        help="All moves that will be hashed",
+    )
+    warnings = fields.Json(
+        compute='_compute_warnings',
+    )
+
+    def new(self, values=None, origin=None, ref=None):
+        res = super().new(values=values, origin=origin, ref=ref)
+        for wizard in res:
+            if not wizard.company_id:
+                wizard.company_id = self.env.company
+            if not wizard.hash_date:
+                wizard.hash_date = wizard.max_hash_date or fields.Date.context_today(self)
+        return res
```

Comment:
Just have the hash date as a stored computed field then

```suggestion
    @api.depends('max_hash_date')
    def _compute_hash_date(self):
        for wizard in self:
            wizard.hash_date = wizard.max_hash_date or fields.Date.context_today(self)
```